portability-transfer/src/main/java/org/datatransferproject/transfer/JobProcessor.java

```diff
@@ -75,11 +75,13 @@ void processJob() {
     boolean success = false;
     UUID jobId = JobMetadata.getJobId();
     monitor.debug(() -> format("Begin processing jobId: %s", jobId));
+    markJobStarted(jobId);
     hooks.jobStarted(jobId);
 
     PortabilityJob job = store.findJob(jobId);
     JobAuthorization jobAuthorization = job.jobAuthorization();
     Preconditions.checkState(jobAuthorization.state() == JobAuthorization.State.CREDS_STORED);
```

Comment:
i think CREDS_STORED should be a pre-condition of job-started.. WDYT of checking CREDS_STORED as a validator to the update in markJobStarted?

that will also do it as part of the same transaction so we may be able to save a transaction

there's an example at https://github.com/google/data-transfer-project/blob/228378c5d92cd5853e26080f4328161f6c154a97/portability-api/src/main/java/org/datatransferproject/api/action/transfer/ReserveWorkerAction.java#L64

Reply:
Good point - WDYT of this?

Reply:
nice! thanks for getting the other place too